com.unity.render-pipelines.universal/ShaderLibrary/Shadows.hlsl

```diff
@@ -343,13 +346,38 @@ half AdditionalLightRealtimeShadow(int lightIndex, float3 positionWS, half3 ligh
     return SampleShadowmap(TEXTURE2D_ARGS(_AdditionalLightsShadowmapTexture, sampler_AdditionalLightsShadowmapTexture), shadowCoord, shadowSamplingData, shadowParams, true);
 }
 
-half GetShadowFade(float3 positionWS)
+half GetMainLightShadowFade(float3 positionWS)
+{
+    // This code computes distance between capsule and point.
+    // Finally distance is used with shadow fade scale and bias to compute shadow fade blend value.
+
+    float3 shadowFadeCenter = _CascadeShadowSplitSphereLast.xyz;
```

Comment:
We need to fade when the shadow gets clipped for all cascades. The shadow can be clipped by the sphere in any cascade level with the current implementation in trunk (this is really a bug with the original stable fit implementation but we intend to keep this functionality going forward for performance/quality reasons).

Reply:
Could you eleborate on this more, I am not sure I got the issue

Reply:
The general problem you are fixing isn't specific to only the last cascade. Every cascade sphere clips the shadow in the corners.